subsys/llext/llext.c

```diff
@@ -18,6 +17,13 @@ LOG_MODULE_REGISTER(llext, CONFIG_LLEXT_LOG_LEVEL);
 
 #include <string.h>
 
+#ifdef CONFIG_MMU_PAGE_SIZE
+#define LLEXT_PAGE_SIZE CONFIG_MMU_PAGE_SIZE
```

Comment:
This only matters if you have userspace enabled, otherwise this does not need to be page aligned right ? This will waste memory when not using userspace.

Reply:
That's not true as regions need to have memory attributes (r/w/x) assigned even without userspace around in this case, particularly the executable bit needs to be settable for the .text region such that its rx, and ideally the .rodata should be ro. Without being able to manipulate the page permissions the loadable module will fail to execute as the executable bit on the allocated or peeked at memory will not be setup correctly.

Reply:
oh sorry, yep that is true. This needs to be page aligned.